### DISCUSSION TITLE
is there a way to make my machine ready for visualization?

### DISCUSSION THREAD

**Discussion by bennyKY** [statelyai/xstate#1875] [Q&A] [answered]
Hi, i have a big machine, that uses other machine. Its written in typescript. Every time i want to visualize my machine i need to remove a lot of typescript related stuff and do many more transformations.
Before im writing a script that takes my .ts file, resolves relevant references to other machines the resides in other files, and generates a .js output that ready to paste in the visualizer, i wanted to check if someone share the struggle and maybe familiar with some (at least partial) solution?

**Comment by davidkpiano** [comment, marked as answer]
Yes: https://xstate.js.org/docs/packages/xstate-inspect/

**Reply by bennyKY** [reply]
@davidkpiano 
Nice! But im using Nodejs :(.
Anything i can do?

**Reply by davidkpiano** [reply]
Yes: https://github.com/davidkpiano/xstate/blob/master/packages/xstate-inspect/examples/server.ts

**Reply by bennyKY** [reply]
Thanks, running this example locally, where can i visualize it?

**Reply by davidkpiano** [reply]
That you will have to implement yourself. `@xstate/inspect` gives you all of the tools/events to make it possible to visualize it however you want: https://xstate.js.org/docs/packages/xstate-inspect/#implementing and the visualizations present on statecharts.io are on the `viz` branch, but there is no documentation.

**Reply by tomByrer** [reply]
@bennyKY I don't know about a full visualization, but for my testings I found it handy to use `nextEvents` to build a button GUI to manually trigger those next events.
